label_studio/ml/models.py

```diff
@@ -258,6 +258,60 @@ def _predict(self, task):
             },
         }
 
+    def _get_predictions_from_ml_backend(self, serialized_tasks: List[Dict]) -> List[Dict]:
+        result = self.api.make_predictions(serialized_tasks, self.project)
+
+        # response validation
+        if result.is_error:
+            logger.error(f'Error occurred: {result.error_message}')
+            return []
+        elif not isinstance(result.response, dict) or 'results' not in result.response:
+            logger.error(f'ML backend returns an incorrect response, it must be a dict: {result.response}')
+            return []
+        elif not isinstance(result.response['results'], list) or len(result.response['results']) == 0:
+            logger.error('ML backend returns an incorrect response, it must be a list with at least one result')
+            return []
+
+        responses = result.response['results']
+
+        predictions = []
+        if len(responses) == 1 and len(serialized_tasks) > 1:
+            # In case ML backend doesn't support batch of tasks, do it one by one
+            # TODO: remove this block after all ML backends will support batch processing
+            logger.warning(
+                f"'ML backend '{self.title}' doesn't support batch processing of tasks, "
+                f'switched to one-by-one task retrieval'
+            )
+            for serialized_task in serialized_tasks:
+                # get predictions per task
+                predictions.extend(self._get_predictions_from_ml_backend([serialized_task]))
+
+            return predictions
+
+        # ML backend supports batch processing
+        for task, response in zip(serialized_tasks, responses):
+            if isinstance(response, dict):
+                # ML backend can return single prediction per task or multiple predictions
+                response = [response]
+
+            # get  all predictions per task
+            for r in response:
+                if 'result' not in r:
+                    logger.error(
+                        f"ML backend returns an incorrect prediction, it should be a dict with the 'result' field:"
+                        f' {r}'
+                    )
+                    continue
+                predictions.append(
+                    {
+                        'task': task['id'],
+                        'result': r['result'],
+                        'score': r.get('score'),
+                        'model_version': r.get('model_version', self.model_version),
```

Comment:
project must be specified: 

```suggestion
                        'model_version': r.get('model_version', self.model_version),
                        'project': task['project']
```

Reply:
isn't it already supported in `Prediction.save()` ?

Reply:
yes, but with warning - 
![image](https://github.com/HumanSignal/label-studio/assets/501780/10f39216-76e5-484a-880c-3d87771406e7)
it seems it is better to set it explicitly